src/GovUk.Education.ExploreEducationStatistics.Data.Services.Tests/ReleaseServiceTests.cs

```diff
@@ -243,6 +249,107 @@ await statisticsDbContext.IndicatorGroup.AddRangeAsync(
             }
         }
 
+        [Fact]
+        public async Task ListSubjects_NoSubjects()
+        {
+            var statsReleaseVersion = new Data.Model.ReleaseVersion();
+
+            var contentReleaseVersion = new ReleaseVersion
+            {
+                Id = statsReleaseVersion.Id,
+            };
+
+            await using var statisticsDbContext = InMemoryStatisticsDbContext();
+            await using var contentDbContext = InMemoryContentDbContext();
+
+            statisticsDbContext.ReleaseVersion.Add(statsReleaseVersion);
+            contentDbContext.ReleaseVersions.Add(contentReleaseVersion);
+
+            var service = BuildReleaseService(
+                contentDbContext: contentDbContext,
+                statisticsDbContext: statisticsDbContext
+            );
+
+            var result = await service.ListSubjects(contentReleaseVersion.Id);
+
+            result.AssertNotFound();
+        }
+
+        [Fact]
+        public async Task ListSubjects_StatsDbHasMissingSubject()
+        {
+            var statisticsReleaseVersion = _fixture.DefaultStatsReleaseVersion()
+                .Generate();
+
+            var releaseSubject1 = _fixture.DefaultReleaseSubject()
+                .WithReleaseVersion(statisticsReleaseVersion)
+                .WithSubject(_fixture.DefaultSubject())
+                .Generate();
+
+            var statisticsDbContextId = Guid.NewGuid().ToString();
+            await using (var statisticsDbContext = InMemoryStatisticsDbContext(statisticsDbContextId))
+            {
+                statisticsDbContext.ReleaseSubject.Add(releaseSubject1);
+                await statisticsDbContext.SaveChangesAsync();
+            }
+
+            var contentReleaseVersion = _fixture.DefaultReleaseVersion()
+                .WithId(statisticsReleaseVersion.Id)
+                .Generate();
+
+            var releaseFile1 = _fixture.DefaultReleaseFile()
+                .WithReleaseVersion(contentReleaseVersion)
+                .WithFile(_fixture.DefaultFile()
+
+                    .WithType(FileType.Data)
+                    .WithSubjectId(releaseSubject1.SubjectId))
+                .Generate();
+
+            var releaseFile2 = _fixture.DefaultReleaseFile()
+                .WithReleaseVersion(contentReleaseVersion)
+                .WithFile(_fixture.DefaultFile()
+                    .WithType(FileType.Data)
+                    .WithSubjectId(Guid.NewGuid()))
+                .Generate();
+
+            var import1 = _fixture.DefaultDataImport()
+                .WithFile(releaseFile1.File)
+                .WithStatus(DataImportStatus.COMPLETE)
+                .Generate();
+
+            var import2 = _fixture.DefaultDataImport()
+                .WithFile(releaseFile2.File)
+                .WithStatus(DataImportStatus.COMPLETE)
+                .Generate();
```

Comment:
You could avoid all of these unnecessary `.Generate()`  by declaring the type rather than using `var`, which will use the implicit operator, e.g.

```csharp
            ReleaseSubject releaseSubject1 = _fixture.DefaultReleaseSubject()
                .WithReleaseVersion(statisticsReleaseVersion)
                .WithSubject(_fixture.DefaultSubject());
```